docs/how-to-build/how-to-build.md

```diff
@@ -866,3 +866,50 @@ ${HM_SDK}/native/build-tools/cmake/bin/cmake -DOHOS_STL=c++_static -DOHOS_ARCH=a
 make -j$(nproc)
 make install
 ```
+
+### Build for ESP32 with cross-compiling
+Download esp-idf sdk
+```shell
+git clone https://github.com/espressif/esp-idf
+cd esp-idf
+git submodule update --init --recursive
+```
+Install esp-idf sdk and configure the environment
+```shell
+sudo sh install.sh
+source export.sh
+```
+Note: python>=3.8, cmake>=3.24.0
+
+Setting up cross-compilation, Create `esp32.toolchain.cmake` in the `ncnn/toolchains` directory, Select the appropriate toolchain in esp-idf.
+```cmake
+set(CMAKE_SYSTEM_NAME freertos)
+set(CMAKE_SYSTEM_PROCESSOR xtensa-esp32)
+
+include($ENV{IDF_PATH}/tools/cmake/toolchain-esp32.cmake)
+
+set(CMAKE_TRY_COMPILE_TARGET_TYPE STATIC_LIBRARY)
+
+set(CMAKE_FIND_ROOT_PATH_MODE_PROGRAM NEVER)
+set(CMAKE_FIND_ROOT_PATH_MODE_LIBRARY ONLY)
+set(CMAKE_FIND_ROOT_PATH_MODE_INCLUDE ONLY)
+set(CMAKE_FIND_ROOT_PATH_MODE_PACKAGE ONLY)
+
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS}" CACHE STRING "c flags")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}" CACHE STRING "c++ flags")
+
+option(NCNN_BUILD_BENCHMARK "" OFF)
+```
```

Comment:
we have in-house toolchain file now, this section is not needed, right ?

Reply:
Right, we can directly use the provided in the project